server/src/main/java/keywhiz/service/resources/automation/v2/BackupResource.java

```diff
@@ -0,0 +1,153 @@
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import javax.ws.rs.GET;
+import javax.ws.rs.InternalServerErrorException;
+import javax.ws.rs.NotFoundException;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.ServiceUnavailableException;
+import keywhiz.KeywhizConfig;
+import keywhiz.api.SecretDeliveryResponse;
+import keywhiz.api.model.AutomationClient;
+import keywhiz.api.model.Group;
+import keywhiz.log.AuditLog;
+import keywhiz.log.Event;
+import keywhiz.service.config.Readonly;
+import keywhiz.service.daos.GroupDAO;
+import keywhiz.service.daos.GroupDAO.GroupDAOFactory;
+import keywhiz.service.daos.SecretController;
+import org.bouncycastle.openpgp.PGPException;
+import org.c02e.jpgpj.Encryptor;
+import org.c02e.jpgpj.FileMetadata;
+import org.c02e.jpgpj.Key;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.time.Instant.now;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static javax.ws.rs.core.MediaType.APPLICATION_OCTET_STREAM;
+import static keywhiz.log.EventTag.GROUP_BACKUP;
+import static org.c02e.jpgpj.CompressionAlgorithm.ZIP;
+import static org.c02e.jpgpj.EncryptionAlgorithm.AES256;
+import static org.c02e.jpgpj.FileMetadata.Format.UTF8;
+import static org.c02e.jpgpj.HashingAlgorithm.Unsigned;
+
+/**
+ * @parentEndpointName automation/v2-backup
+ * @resourceDescription Automation endpoints to manage groups
+ */
+@Path("/automation/v2/backup")
+public class BackupResource {
+  private static final Logger logger = LoggerFactory.getLogger(BackupResource.class);
+
+  private final GroupDAO groupDAO;
+  private final SecretController secretController;
+  private final ObjectMapper objectMapper;
+  private final KeywhizConfig config;
+  private final AuditLog auditLog;
+
+  @Inject
+  public BackupResource(
+      GroupDAOFactory groupDAOFactory,
+      @Readonly SecretController secretController,
+      ObjectMapper objectMapper,
+      KeywhizConfig config,
+      AuditLog auditLog) {
+    this.groupDAO = groupDAOFactory.readonly();
+    this.secretController = secretController;
+    this.objectMapper = objectMapper;
+    this.config = config;
+    this.auditLog = auditLog;
+  }
+
+  /**
+   * Backup all secrets for a given group. Returns an encrypted encrypted to
+   * the backup key in the main configuration file. Only accessible to automation clients.
+   *
+   * @excludeParams automationClient
+   * @param name Group name
+   *
+   * @return Encrypted archive
+   */
+  @Timed @ExceptionMetered
+  @GET @Path("{group}")
+  @Produces(APPLICATION_OCTET_STREAM)
+  public byte[] backup(
+      @Auth AutomationClient automationClient,
```

Comment:
We could add a flag to the automation clients to control which ones can hit the backup endpoint.